h2/utilities.py

```diff
@@ -197,10 +197,10 @@ def validate_headers(headers, hdr_validation_flags):
     headers = _reject_surrounding_whitespace(
         headers, hdr_validation_flags
     )
-    headers = _reject_te(
+    headers = _strip_connection_header(
```

Comment:
I moved the stripping logic into `normalize_outbound_headers`, and this now works as anticipated. As for the separate pipeline, it looks like both the questions posed involve the receipt of invalid headers rather than sending. With this normalization only running for outbound headers, we will properly raise a ProtocolError if we receive a malformed message. 

> An endpoint MUST NOT generate an HTTP/2 message containing connection-specific header fields;

I feel like stripping the headers satisfies this requirement but I can understand a situation where someone may want to catch the exception and possibly change the request workflow. If we do decide to go this route, how does a flag like `normalize_transformed_connection_headers` and a separate normalization function sound?